master/internal/api_project.go

```diff
@@ -308,64 +297,25 @@ func (a *apiServer) getProjectColumnsByID(
 		}
 	}
 
-	// Get metrics columns
-	metricNames, err := a.getProjectMetricsNames(ctx, curUser, p)
-	if err != nil {
-		return nil, err
-	}
-
-	for _, mn := range metricNames {
-		columns = append(columns, &projectv1.ProjectColumn{
-			Column:   fmt.Sprintf("validation.%s", mn),
-			Location: projectv1.LocationType_LOCATION_TYPE_VALIDATIONS,
-			Type:     projectv1.ColumnType_COLUMN_TYPE_NUMBER,
-		})
-	}
-
 	return &apiv1.GetProjectColumnsResponse{
 		Columns: columns,
 	}, nil
 }
 
-func (a *apiServer) getProjectMetricsNames(
-	ctx context.Context, curUser model.User, project *projectv1.Project,
-) ([]string, error) {
-	metricNames := []struct {
-		Vname       []string
-		WorkspaceID int
-	}{}
-
-	metricQuery := db.Bun().
-		NewSelect().
-		TableExpr("exp_metrics_name").
-		TableExpr("LATERAL json_array_elements_text(vname) AS vnames").
-		ColumnExpr("array_to_json(array_agg(DISTINCT vnames)) AS vname").
-		ColumnExpr("?::int as workspace_id", project.WorkspaceId).
-		Where("project_id = ?", project.Id)
-
-	metricQuery, err := exputil.AuthZProvider.Get().FilterExperimentsQuery(
-		ctx,
-		curUser,
-		project,
-		db.Bun().NewSelect().TableExpr("(?) AS subq", metricQuery),
-		[]rbacv1.PermissionType{rbacv1.PermissionType_PERMISSION_TYPE_VIEW_EXPERIMENT_ARTIFACTS},
-	)
-	if err != nil {
-		return nil, err
-	}
-
-	err = metricQuery.Scan(ctx, &metricNames)
-	if err != nil {
-		return nil, errors.Wrapf(
-			err, "error fetching metrics names for project (%d) from database", project.Id)
+func parseMetricsType(metricType interface{}) projectv1.ColumnType {
```

Comment:
why does this take `interface{}` and then immediately cast to `string`?

Reply:
My bad, fixed